### PR TITLE
Ensure view build progress

### PR DESCRIPTION
**Pull request by davisp**
## Overview

View indexers create conflicts when reading from the changes feed which can prevent the indexer from making progress on databases that are being updated continuously.

I'm marking this as a draft because it doesn't yet address the situation of a single doc being updated repeatedly and there's no tests proving it works either. (Second time's a charm!)

## Testing recommendations

I should add a test.

## Checklist

- [x] Code is written and works correctly
- [ ] Changes are covered by tests
- [x] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

**Comment by nickva** [comment]
I noticed a few exceptions when running unit tests:

```
 CRASH REPORT Process  (<0.671.0>) with 0 neighbors crashed with reason: no function clause matching erlfdb:get_ss({erlfdb_snapshot,{erlfdb_transaction,#Ref<0.2213159554.4118937601.149766>}}, <<"�/metadataVersion">>)(line:317) at fabric2_fdb:check_metadata_version/1(line:1316) <= fabric2_fdb:ensure_current/2(line:1897) <= fabric2_fdb:get_last_change/1(line:1100) <= fabric2_fdb:execute_transaction/3(line:1960) <= erlfdb:do_transaction/2(line:663) <= fabric2_fdb:do_transaction/2(line:171) <= fabric2_fdb:transactional/2(line:142); initial_call: {couch_views_indexer,init,[]}, ancestors: [couch_views_server,couch_views_sup,<0.662.0>], message_queue_len: 0, messages: [], links: [<0.666.0>], dictionary: [{'$couch_jobs_current',{erlfdb_transaction,#Ref<0.2213159554.4118937601.149649>}},...], trap_exit: false, status: running, heap_size: 1598, stack_size: 27, reductions: 16199
```

It looks like we'd need a no-op or passthrough case for when we get a snapshot of a snapshot handle.

**Comment by davisp** [comment]
Huh, I thought I'd added that at one point. Good catch.

**Comment by davisp** [comment]
Ah derps, I had that locally but forgot to create a spearate PR for erlfdb since its a separate repo. Thanks for the hand there!

**Review by nickva — APPROVED** [review]

**Comment by nickva** [comment]
@davisp Maybe add a few tests if you'd like to get some coverage for the new bits

**Comment by davisp** [comment]
Yap. Forgot I hadn't done that bit yet. Doing now before merging.

File: src/fabric/test/fabric2_snapshot_tests.erl [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+bump_view_size(TxDb) ->
+    #{
+        tx := Tx,
+        db_prefix := DbPrefix
+    } = TxDb,
+    
+    DbTuple = {?DB_STATS, <<"sizes">>, <<"views">>},
+    DbKey = erlfdb_tuple:pack(DbTuple, DbPrefix),
+    erlfdb:add(Tx, DbKey, 0).
+
```
Comment (nickva) [inline src/fabric/test/fabric2_snapshot_tests.erl:122]: Add an extra line between functions

Comment (davisp) [reply src/fabric/test/fabric2_snapshot_tests.erl:126]: Done.

**Review by nickva — APPROVED** [review]

**Pull request merged**